src/AntiMattr/MongoDB/Migrations/Version.php

```diff
@@ -276,13 +284,24 @@ public function executeScript(Database $db, $file)
         return $result;
     }
 
-    public function markMigrated()
+    /**
+     * markMigrated.
+     *
+     * @param bool $replay This is a replayed migration, do an update instead of an insert
+     */
+    public function markMigrated($replay = false)
     {
         $this->configuration->createMigrationCollection();
         $collection = $this->configuration->getCollection();
 
         $document = array('v' => $this->version, 't' => $this->createMongoTimestamp());
-        $collection->insert($document);
+
+        if ($replay) {
+            $query = array('v' => $this->version);
+            $collection->update($query, $document);
```

Comment:
What happens when we use "replay" and there was no previous execution of the this version ?